File: backend/rails/yoember/app/models/library.rb
class Library < ApplicationRecord
  before_save :downcase_fields

  validates_presence_of :name, :address, :phone

  self.per_page = 10

  private

  def downcase_fields
    self.name = name.downcase
    self.address = address.downcase
    self.phone = phone.downcase
  end
end

File: backend/phoenix/yoember/lib/yoember/libraries/library.ex
defmodule Yoember.Libraries.Library do
  use Ecto.Schema
  import Ecto.Changeset
  alias Yoember.Libraries.Library

  schema "libraries" do
    field :address, :string
    field :name, :string
    field :phone, :string

    timestamps()
  end

  @doc false
  def changeset(%Library{} = library, attrs) do
    library
    |> cast(attrs, [:name, :address, :phone])
    |> validate_required([:name, :address, :phone])
  end
end
